### PR TITLE
[Kotlin] Add optional parameters to the method of api

### PR DESCRIPTION
**Pull request by keijumt**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh`, `./bin/security/{LANG}-petstore.sh` and `./bin/openapi3/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR
Please see #2012 

(details of the change, additional tests that have been done, reference to the issue for tracking, etc)

### CODE REVIEW

**Comment by wing328** [comment]
cc @jimschubert (2017/09), @dr4ke616 (2018/08)

**Comment by wing328** [comment]
@keijumt please you please run the kotlin petstore scripts (./bin/kotlin*) or batch files (bin\windows\kotlin*) to update the petstore samples so that the CIs can verify the change?

File: samples/client/petstore/kotlin-string/src/main/kotlin/org/openapitools/client/apis/PetApi.kt [hunk 1]
```diff
@@ -56,7 +56,7 @@ class PetApi(basePath: kotlin.String = "http://petstore.swagger.io/v2") : ApiCli
     * @param apiKey  (optional, default to null)
     * @return void
     */
-    fun deletePet(petId: kotlin.Long, apiKey: kotlin.String) : Unit {
+    fun deletePet(petId: kotlin.Long, apiKey: kotlin.String? = null) : Unit {
```
Comment (wing328) [inline samples/client/petstore/kotlin-string/src/main/kotlin/org/openapitools/client/apis/PetApi.kt]: @keijumt overall the change looks good.

Ideally should ` = null` be omitted (and only show the non-null default value)?

Comment (keijumt) [reply samples/client/petstore/kotlin-string/src/main/kotlin/org/openapitools/client/apis/PetApi.kt]: @wing328 I Modified to omit `= null`.

**Comment by sylvainmoindron** [comment]
i ran into a bug with this commit should I open an issue or comment here now that it's merged ?
this PR add ? in the api class for optional parameter but not in the ApiService interface

**Pull request merged**